reactVR_course/solutions/Task6/task3and4/Game.vr.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import {Animated, View, NativeModules} from 'react-vr'
+
+import Intro3DModel as DeathStar from './Intro3DModel.vr.js';
+
+import ComponentGenerator, {getRandomCoordinates, randomComponents} from '../../helpers/ComponentGenerator';
+
+const NUM_COMPONENTS = 10;
+const Y_POSITION = 8;
+const GAME_TIME = 30;
+
+export default class ShapeGenerator extends React.Component {
+  constructor() {
+    super();
+
+    this.startAnimation = this.startAnimation.bind(this);
+    this.getShapesNextState = this.getShapesNextState.bind(this);
+    this.onShootShape = this.onShootShape.bind(this);
+
+    const yPosition = new Animated.Value(Y_POSITION);
+    this.state = {
+      globalYPosition: yPosition,
+      time: GAME_TIME,
+      Obstacles: () => <ComponentGenerator yPosition={yPosition} numberOfComponents={NUM_COMPONENTS} onClick={this.onShootShape(-1)}/>,
+      score: 0
+    }
+  }
+
+  componentDidMount() {
+    this.startAnimation();
+  }
+
+  startAnimation() {
+    Animated.timing(this.state.globalYPosition, {
+      duration: 15000,
+      toValue: -Y_POSITION
+    }).start((o) => {
+      if (o.finished) {
+        this.setState(this.getShapesNextState());
+      }
+    });
+  }
+
+  getShapesNextState() {
+    this.state.globalYPosition.setValue(Y_POSITION);
+    this.startAnimation();
+
+    const {globalYPosition} = this.state;
+    return {
+      Obstacles: () => <ComponentGenerator yPosition={globalYPosition} numberOfComponents={NUM_COMPONENTS} onClick={this.onShootShape(-1)}/>,
+    };
+  }
+
+  onShootShape(clickPoint) {
+    return () => {
+      this.setState({score: this.state.score + clickPoint});
+    }
+  }
+
+  render() {
+    const {Obstacles, deathStarPosition, score, globalYPosition} = this.state;
+    NativeModules.DomOverlayModule.openOverlay({time: globalTimer, score: score});
+      return (
+        <View>
+          <Obstacles/>
+          <DeathStar coordinates={deathStarPosition} onClick={this.onShootShape(1)}/>
```

Comment:
Vi burde egentlig skrive om denne sånn at den bruker Intro3DModel istedenfor DeathStar, siden det er den de har jobbet med hele veien. Men det er ikke så viktig til i dag :)

Reply:
Så plutselig at det faktisk er Intro3DModel som blir brukt, er det ikke litt forvirrende at vi plutselig endrer navn på den?
Og fungerer dette egentlig, fordi Intro3DModel bruker vel ikke de propsene som blir sendt inn?

Reply:
Kan endre på det, fordi det er nok forvirrende.

Reply:
Deathstar bruker koordinater som er tatt inn via props, men det gjør ikke Intro3Dmodel. 

Vi kan evt. ha en mellomoppgave for å gjøre om Intro3DModel så den blir lik som deathstar? Eller hva tenker dere?

Reply:
Det trengs ikke koordinater som props. Det er kun noe som henger igjen fra en gammel commit :)